optional reset GPIO for TLE9104

benediktibk opened the pull request:
In some hardware designs it might happen that the reset signal for the TLE9104 is not used only for this purpose, but instead for instance to reset other devices at the same time. For such a hardware design it is then necessary to make the reset GPIO optional. The reset will have to be triggered earlier on.

benediktibk committed:
drivers: gpio: make reset of TLE9104 optional

In some hardware designs it might happen that the reset signal
for the TLE9104 is not used only for this purpose, but instead for
instance to reset other devices at the same time. For such a hardware
design it is then necessary to make the reset GPIO optional. The reset
will have to be triggered earlier on.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

henrikbrixandersen commented on drivers/gpio/gpio_tle9104.c:
```diff
@@ -433,16 +433,18 @@ static int tle9104_init(const struct device *dev)
-		return result;
-	}
+	if (config->gpio_reset.port != NULL) {
+		result = gpio_pin_configure_dt(&config->gpio_reset, GPIO_OUTPUT_ACTIVE);
```
Looks like this is missing a call to `gpio_is_ready_dt()`?

benediktibk replied:
Good point, fixed it.

benediktibk committed:
drivers: gpio: use gpio_is_ready_dt in TLE9104

Use gpio_is_ready_dt in the driver for the TLE9104 before
actually using the GPIO.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

henrikbrixandersen commented on drivers/gpio/gpio_tle9104.c:
```diff
@@ -408,8 +408,8 @@ static int tle9104_init(const struct device *dev)
 
-		if (!device_is_ready(current->port)) {
-			LOG_ERR("control GPIO %s is not ready", current->port->name);
+		if (!gpio_is_ready_dt(current)) {
+			LOG_ERR("%s: control GPIO is not ready", dev->name);
```
Were the changes in what is logged intentional? Before, the log would contain the name of the GPIO controller, which was not ready. Now it contains the name of the tle9104 device, for which a GPIO controller wasn't ready.

benediktibk replied:
Yes, this change was intentional. Because gpio_is_ready_dt can be called on a port which is NULL. Which would cause a slight oops in the log statement.

MaureenHelm commented:
@ifyall please revisit
